### PR TITLE
JNG-5600 Add generator

### PR DESCRIPTION
**Pull request by robertcsakany**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5600" title="JNG-5600" target="_blank">Sub-taskJNG-5600</a>  Create model generator engine / maven plugin for JSL
  </td></table>
  <br />

### CODE REVIEW

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/runtime/JslParser.java [hunk 1]
```diff
@@ -214,4 +220,19 @@ private static void collectImportedModelDeclaration(ModelDeclaration modelDeclar
...
     }
 
+    public static Set<ModelDeclaration> getRootModelDeclarations(Collection<ModelDeclaration> modelDeclarations) {
+        Set<String> imprtedModels = new HashSet<>();
+        modelDeclarations.stream().forEach(m -> collectImortedModels(m, imprtedModels));
+        return modelDeclarations.stream().filter(m -> !imprtedModels.contains(m.getName())).collect(Collectors.toSet());
+    }
+
+
+    private static void collectImortedModels(ModelDeclaration modelDeclaration, Set<String> referenced) {
```
Comment (gaborflorian) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/runtime/JslParser.java]: I think it is `collectImportedModels`

File: generator-engine/pom.xml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                                    <directory>src/main/epsilon</directory>
+                                    <filtering>false</filtering>
+                                </resource>
+                            </resources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!--plugin>
```
Comment (noherczeg) [inline generator-engine/pom.xml]: Do we need this?

File: generator-engine/src/main/java/hu/blackbelt/judo/meta/jsl/generator/engine/BufferedSlf4jLogger.java [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+import org.slf4j.Marker;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Set;
+
+@Slf4j
+public class BufferedSlf4jLogger implements Logger, Closeable {
```
Comment (noherczeg) [inline generator-engine/src/main/java/hu/blackbelt/judo/meta/jsl/generator/engine/BufferedSlf4jLogger.java:35]: This logger caused issues in jsl2ui, do we need it here?

File: generator-maven-plugin-test/pom.xml [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                            <goal>checksum</goal>
+                        </goals>
+
+                        <configuration>
+                            <sources>${basedir}/target/classes/jsl</sources>
+                            <destination>${basedir}/target/test-classes/jsl/file</destination>
+                        </configuration>
+                    </execution>
+
+                    <!--
```
Comment (noherczeg) [inline generator-maven-plugin-test/pom.xml]: What about this?

File: generator-maven-plugin/README.adoc [hunk 5]
```diff
@@ -0,0 +1,286 @@
+JUDO Tatami JSL Archetype maven plugin
+======================================
+
+This plugin manages and executes generators for JUDO JSL models.
+
+It generates text based codes / reports from JSL model with the given templates.
+
+== Requirements
+
+- Maven 3.8.3 and Java 11
```
Comment (noherczeg) [inline generator-maven-plugin/README.adoc:10]: 3.9, 17

File: generator-maven-plugin/pom.xml [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+	<artifactId>judo-jsl-generator-maven-plugin</artifactId>
+	<packaging>maven-plugin</packaging>
+	<name>Judo :: JSL :: Model :: Generator :: Maven :: Plugin</name>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-plugin-plugin</artifactId>
+				<version>3.8.2</version>
```
Comment (noherczeg) [inline generator-maven-plugin/pom.xml:23]: 3.9?

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/runtime/JslParser.java [hunk 7]
```diff
@@ -214,4 +220,19 @@ private static void collectImportedModelDeclaration(ModelDeclaration modelDeclar
         }
     }
 
+    public static Set<ModelDeclaration> getRootModelDeclarations(Collection<ModelDeclaration> modelDeclarations) {
+        Set<String> imprtedModels = new HashSet<>();
```
Comment (noherczeg) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/runtime/JslParser.java]: imported?

**Review by noherczeg — CHANGES_REQUESTED** [review]

**Pull request merged**